pkg/operator/apis/monitoring/v1alpha1/types.go

```diff
@@ -679,12 +712,51 @@ func (cm *ClusterPodMonitoring) endpointScrapeConfig(index int) (*promconfig.Scr
 		}
 	}
 
-	// Set a clean namespace, job, and instance label that provide sufficient uniqueness.
-	relabelCfgs = append(relabelCfgs, &relabel.Config{
-		Action:       relabel.Replace,
-		SourceLabels: prommodel.LabelNames{"__meta_kubernetes_namespace"},
-		TargetLabel:  "namespace",
-	})
+	metadataLabels := map[string]struct{}{}
+	// The metadata list must be always set in general but we allow the null case
+	// for backwards compatibility. In that case we must always add the namespace label.
+	if cm.Spec.TargetLabels.Metadata == nil {
+		metadataLabels = map[string]struct{}{
+			"namespace": struct{}{},
+		}
+	} else {
+		for _, l := range *cm.Spec.TargetLabels.Metadata {
+			if allowed := []string{"namespace", "pod", "container", "node"}; !containsString(allowed, l) {
+				return nil, errors.Errorf("metadata label %q not allowed, must be one of %v", l, allowed)
+			}
+			metadataLabels[l] = struct{}{}
+		}
+	}
+	if _, ok := metadataLabels["namespace"]; ok {
```

Comment:
@fabxc - was this missing from the `prometheus_target` schema before? i.e. if a user configured a ClusterPodMonitoring for scraping, was the associated `prometheus_target.namespace` empty in the metric backend?